src/app/loans/loans-view/loan-account-actions/loan-account-actions.component.ts

```diff
@@ -9,11 +13,18 @@ import { ActivatedRoute, Router } from '@angular/router';
 export class LoanAccountActionsComponent {
 
   /** flag object to store possible actions and render appropriate UI to the user */
-  actions: { close: boolean } = { close: false };
+  actions: { close: boolean, undo_approval: boolean } = { close: false, undo_approval: false };
 
+  /**
+   * @param router Router.
+   * @param route Activated Route.
+   */
   constructor(private router: Router,
     private route: ActivatedRoute) {
     this.route.params.subscribe(params => {
+      if (params['action'].indexOf('-') > 0) {
+        this.actions[params['action'].replace('-', '_')] = true;
```

Comment:
We are following a specific naming convention for all the buttons. Please follow that.

Reply:
Here in object you cannot use '-'. So I used a '_' instead

Reply:
Okay, get it. Thanks :)